src/workflows-app/components/SubmitWorkflowModal.test.ts

```diff
@@ -0,0 +1,385 @@
+import {
+  mockAzureApps,
+  mockAzureWorkspace,
+  mockCollaborativeAzureApps,
+  mockCromwellRunner,
+  runSetInputDef,
+  runSetOutputDefFilled,
+} from 'src/workflows-app/utils/mock-responses';
+
+jest.mock('src/libs/ajax');
+
+type AjaxContract = ReturnType<typeof Ajax>;
+
+jest.mock('src/libs/notifications.js');
+
+jest.mock('src/libs/config', () => ({
+  ...jest.requireActual('src/libs/config'),
+  getConfig: jest.fn().mockReturnValue({}),
+}));
+
+jest.mock('src/libs/nav', () => ({
+  getCurrentUrl: jest.fn().mockReturnValue(new URL('https://app.terra.bio')),
+  goToPath: jest.fn(),
+}));
+
+jest.mock('src/libs/state', () => ({
+  ...jest.requireActual('src/libs/state'),
+  getTerraUser: jest.fn().mockReturnValue({ id: 'foo' }),
+}));
+
+jest.mock('src/libs/utils', () => ({
+  ...jest.requireActual('src/libs/utils'),
+  poll: jest.fn(async (fn) => {
+    // remove delay
+    while (true) {
+      const r = await fn();
+      if (!r.shouldContinue) {
+        return r.result;
+      }
+    }
+  }),
+}));
+
+jest.mock('src/components/Modal', () => {
+  const mockModal = jest.requireActual('src/components/Modal.mock');
+  return mockModal.mockModalModule();
+});
+
+jest.mock('src/libs/ajax/metrics/useMetrics', () => ({
+  ...jest.requireActual('src/libs/ajax/metrics/useMetrics'),
+  useMetricsEvent: jest.fn(() => ({ captureEvent: jest.fn() })),
+}));
+
+jest.mock('src/libs/feature-previews', () => ({
+  ...jest.requireActual('src/libs/feature-previews'),
+  isFeaturePreviewEnabled: jest.fn(),
+}));
+
+const submitModalProps = {
+  method: methodDataWithVersions.methods[0],
+  methodVersion: methodDataWithVersions.methods[0].method_versions[0],
+  recordType: 'FOO',
+  selectedRecords: { FOO1: 'bar' },
+  inputDefinition: runSetInputDef,
+  outputDefinition: runSetOutputDefFilled,
+  callCachingEnabled: true,
+  onDismiss: jest.fn(),
+  name: 'test-azure-ws-name',
+  namespace: 'test-azure-ws-namespace',
+  workspace: mockAzureWorkspace,
+};
+
+const submitModalPropsReader = {
+  ...submitModalProps,
+  workspace: {
+    ...mockAzureWorkspace,
+    canCompute: false,
+  },
+};
+
+const postRunSetPayload = expect.objectContaining({
+  method_version_id: methodDataWithVersions.methods[0].method_versions[0].method_version_id,
+  workflow_input_definitions: runSetInputDef,
+  workflow_output_definitions: runSetOutputDefFilled,
+  wds_records: {
+    record_type: 'FOO',
+    record_ids: ['FOO1'],
+  },
+  call_caching_enabled: true,
+});
+
+describe('SubmitWorkflowModal submitting to cromwell', () => {
+  beforeEach(() => {
+    workflowsAppStore.reset();
+    asMockedFn(getConfig).mockReturnValue({
+      wdsUrlRoot: 'https://lz-abc/wds-abc-c07807929cd1/',
+      cbasUrlRoot: 'https://lz-abc/terra-app-abc/cbas',
+      cromwellUrlRoot: 'https://lz-abc/terra-app-abc/cromwell',
+    });
+    asMockedFn(isFeaturePreviewEnabled).mockImplementation((_id) => false);
+  });
+
+  it('should allow submit to cromwell by workspace creator', async () => {
```

Comment:
Are these tests similar enough to warrant a table-style or injection style test consolidation? I could imagine a table of test cases like 
| Feature flag | workspace user  | Submit enabled   |
|---|---|---|
|  No  | creator  | Yes  | 
|  No  | owner  | No  | 
|  No  | writer  | No  | 
|  No  | reader  | No  | 
|  Yes  | creator  | Yes  | 
|  Yes  | admin  | Yes  | 
|  Yes  | writer  | Yes  | 
|  Yes  | reader  | No  | 

You could also add on the "Cromwell App already running" as another case if you were feeling completionist.

I _think_ that would give you 8 (or 16?) test cases for (almost) the same amount of written code as a current single case